03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,22 +1,64 @@
 
 const Task = require('../models/Tasks')
+const asyncWrapper = require('../middleware/async')
+const {createCustomError} = require('../errors/custom-error')
 
-const getAllTasks = (req , res)=>{  
-res.send('all item from the file')
-}
+const getAllTasks = asyncWrapper(async (req , res)=>{  
+    
+        const tasks= await Task.find({})
+        res.status(200).json({tasks})
+        //res.status(200).json({tasks , amount:tasks.length})
+        //res.status(200).json({status: 'success',  data:{tasks, ndHits:tasks.length}})
+    
+})
 
-const createTask= async (req, res) =>{
+const createTask= asyncWrapper(async (req, res) =>{
     const task = await Task.create(req.body)
     res.status(201).json({task})
-}
-const getTask= (req, res) =>{
-    res.json({id: req.param.id})
-}
-const updateTask= (req, res) =>{
-    res.send('update task')
-}
-const deleteTask= (req, res) =>{
-    res.send('delete task')
-}
+})
+
+const getTask= asyncWrapper(async (req, res , next) =>{
+   
+        const{id:taskID}=req.params
+        
+ 
+    const task=await Task.findOne({_id:taskID});
+    if(!task){
+        const err= new Error('not found') ;
+        err.status=404 ;
+        return next(createCustomError('No task with id: ${taskID}'),404) ;
```

Comment:
Need to use back quotes`` for template literals.
Also, the end parathesis ')' was before the 404 parameter. This causes an error in the errorHandlerMiddleware code because err.statusCode was undefined.
```suggestion
return next(createCustomError(`No task with id: ${taskID}`,404) );
```